rfcs/20200817-jit-code-generation-for-aarch64/README.md

```diff
@@ -0,0 +1,256 @@
+`XED.cmake` which will only impact AArch64 builds. 
+[We modified just a few lines](https://github.com/fujitsu/xed/commit/92ee2830ca48fed76baa0a36e7d6ef8a3afe5741) 
+of the original xed sources and disclosed as 
+[fujitsu/xed](https://github.com/fujitsu/xed/tree/fjdev "fujitsu/xed"). 
+
+### Adding primitive entry to src/cpu/cpu_\*_list.cpp
+
+As JIT code for AArch64 is ready to be generated at this point, primitives are 
+added to `src/cpu/cpu_*_list.cpp` as follows not only for convolution but also 
+eltwise, batch_normalization, pooling, etc. This proposal is the same as the 
+proposal in 
+[#795](https://github.com/oneapi-src/oneDNN/pull/795) and the implementation 
+for x64.
+
+```
+static const std::map<conv_impl_key_t, std::vector<pd_create_f>> impl_list_map {
+    // FWD fp
+    {{forward, f32, f32, f32}, {
+        CPU_INSTANCE_X64(jit_avx512_common_dw_convolution_fwd_t)
+        CPU_INSTANCE_X64(jit_avx512_common_1x1_convolution_fwd_f32_t)
+        CPU_INSTANCE_X64(jit_avx512_core_f32_wino_conv_2x3_fwd_t)
+        CPU_INSTANCE_X64(jit_avx512_core_f32_wino_conv_4x3_fwd_t)
+        CPU_INSTANCE_X64(jit_avx512_common_convolution_winograd_fwd_t)
+        CPU_INSTANCE_X64(jit_avx512_common_convolution_fwd_t<f32>)
+        CPU_INSTANCE_X64(jit_avx2_dw_convolution_fwd_t)
+        CPU_INSTANCE_X64(jit_avx2_1x1_convolution_fwd_t)
+        CPU_INSTANCE_X64(jit_sse41_dw_convolution_fwd_t)
+        CPU_INSTANCE_X64(jit_sse41_1x1_convolution_fwd_t)
+        CPU_INSTANCE_X64(jit_avx2_convolution_fwd_t)
+        CPU_INSTANCE_X64(jit_sse41_convolution_fwd_t)
+        CPU_INSTANCE_AARCH64(jit_aarch64_sve_512_common_dw_convolution_fwd_t)         <- Added
+        CPU_INSTANCE_AARCH64(jit_aarch64_sve_512_common_1x1_convolution_fwd_f32_t)    <- Added
+        CPU_INSTANCE_AARCH64(jit_aarch64_sve_512_core_f32_wino_conv_2x3_fwd_t)        <- Added
+        CPU_INSTANCE_AARCH64(jit_aarch64_sve_512_core_f32_wino_conv_4x3_fwd_t)        <- Added
+        CPU_INSTANCE_AARCH64(jit_aarch64_sve_512_common_convolution_winograd_fwd_t)   <- Added
+        CPU_INSTANCE_AARCH64(jit_aarch64_sve_512_common_convolution_fwd_t<f32>)       <- Added
+        CPU_INSTANCE(gemm_convolution_fwd_t)
+        CPU_INSTANCE(ref_convolution_fwd_t<f32>)
+        CPU_INSTANCE(ref_fused_convolution_fwd_t)
+        nullptr,
+    }},
+```
+
+The figure below shows the types of SIMD instructions of AArch64. 
+AArch64 has SIMD & FP(Floating point) instructions as 128-bit long SIMD 
+instructions, and SVE instructions as more than or equal to 128-bit SIMD 
+instructions.SVE and SIMD & FP instructions have relationship where SVE 
+extends SIMD & FP instructions. Bit width of SVE instructions is implementation 
+dependent. AArch64 specification allows for 128 *N* (*N* = 1, 2, ..., 16).
+
+![SIMD instructions of AArch64](aarch64_isa.png)
+
+A primitive that uses only SIMD & FP instructions includes `aarch64_simdfp` in 
+its name.
```

Comment:
Is SIMD & FP the same as Neon?

Reply:
Depending on the definition of NEON, my understanding is that SIMD&FP is not the same as NEON.
NEON is an instruction set that runs at 32-bit state on Armv8 or Armv7 architectures, i.e. AArch32.
On the other hand, SIMD&FP is an instruction set that runs at 64-bit state on Armv8 architectures, i.e. AArch64.
(We can see the words "SIMD&FP" in AArch64 part of the Armv8 reference manual, but not the word "NEON".
In ArmCL, the source code for both NEON and SIMD&FP is in a directory named "NEON".)

Reply:
Oh, I see. Thanks for the elaboration!
I was misled a little by the following [guide](https://developer.arm.com/architectures/instruction-sets/simd-isas/neon/neon-programmers-guide-for-armv8-a/introducing-neon-for-armv8-a/fundamentals-of-armv8-neon-technology).

Reply:
Hi,

The reference manual does indeed talk about 'Advanced SIMD and Floating Point', but from what I've seen it's more common to refer to this as 'Advanced SIMD'. As I understand it, this was an optional extension to v7, but an integral part of v8. Accessible in both 32bit and 64 bit execution modes.
Arm's implementation of this is called Neon (https://developer.arm.com/architectures/instruction-sets/simd-isas/neon) and, colloquially, you would commonly see v8a SIMD features referred to as 'Neon' - certainly in any Arm Developer Guides, but often in third party materials too.
So @emfomenk, @kawakami-k, I think you're both right!